napari/_qt/layer_controls/qt_image_controls_base.py

```diff
@@ -259,3 +251,33 @@ def reset():
             range_btn.setFixedWidth(65)
             range_btn.clicked.connect(layer.reset_contrast_limits_range)
             self._layout.addWidget(range_btn, alignment=Qt.AlignBottom)
+
+
+def range_to_decimals(range_, dtype):
+    """Convert a range to decimals of precision.
+
+    Parameters
+    ----------
+    range : tuple
+        Slider range, min and then max values.
+    dtype : np.dtype
+        Data type of the layer. Integers layers are given integer.
+        step sizes.
+
+    Returns
+    -------
+    int
+        Decimals of precision.
+    """
+
+    if np.issubdtype(dtype, np.integer):
```

Comment:
The following works to support `tensorstore.dtype` objects  [tensorstore API page here](https://google.github.io/tensorstore/python/api/tensorstore.dtype.html#tensorstore.dtype))

```suggestion
 
    if hasattr(dtype, 'numpy_dtype'):
        # retrieve the corresponding numpy.dtype from a tensorstore.dtype
        dtype = dtype.numpy_dtype

    if np.issubdtype(dtype, np.integer):
```